src/panel/app_panel.js

```diff
@@ -81,6 +81,20 @@ export default class App {
     const urlWithoutHash = window.location.href.split('#')[0];
     window.history.replaceState(window.history.state, null, `${urlWithoutHash}#${hash}`);
   }
-}
-
 
+  /**
+   * Go to the previous application state using history.back()
+   * If no previous state is available, use replaceState as a fallback.
+   * @param {Object} fallback
+   * @param {Object} fallback.relativeUrl - The relativeUrl to fallback to if history.back is unavailable
+   * @param {Object} fallback.state - The state to fallback to if history.back is unavailable
+   */
+  back({ relativeUrl = '/', state = {} }) {
+    if (history.state !== null) {
```

Comment:
This is rather fragile.
This can only ensure the app has been loaded into a new tab without previous history.

Reply:
> This can only ensure the app has been loaded into a new tab without previous history.

This doesn't ensure that, because we don't have access to previous history entries for security reason. It would be the same if coming from another website which has a link to an "internal" state of Qwant Maps.

It works because our internal router always put at least an empty object as state, so a `null` means we didn't use `navigateTo` to create the current history state.

This is fragile indeed because it relies on the fact that we don't add anything to the state when using `updateUrl` while loading the results AND the state keeps its `null` value, ignoring the default of `{ }` specified in the `navigateTo` arguments because it works only for `undefined`.

On the other hand, it works for our current bug :)
But maybe we could find a way to make it more fragile… Maybe by always setting a history state value in `navigateTo` used in `replace` mode, and check it instead of `null`.